modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonoidKLaws.kt

```diff
@@ -1,41 +1,44 @@
 package arrow.test.laws
 
 import arrow.Kind
+import arrow.core.extensions.eq
 import arrow.core.extensions.list.foldable.fold
+import arrow.test.generators.GenK
 import arrow.typeclasses.Applicative
 import arrow.typeclasses.Eq
+import arrow.typeclasses.EqK
 import arrow.typeclasses.MonoidK
 import io.kotlintest.properties.Gen
 import io.kotlintest.properties.forAll
 
 object MonoidKLaws {
 
+  fun <F> laws(SGK: MonoidK<F>, GENK: GenK<F>, EQK: EqK<F>): List<Law> =
+    laws(SGK, GENK.genK(Gen.int()), EQK.liftEq(Int.eq()))
+
   fun <F> laws(SGK: MonoidK<F>, AP: Applicative<F>, EQ: Eq<Kind<F, Int>>): List<Law> =
-    SemigroupKLaws.laws(SGK, AP, EQ) + listOf(
-      Law("MonoidK Laws: Left identity") { SGK.monoidKLeftIdentity(AP::just, EQ) },
-      Law("MonoidK Laws: Right identity") { SGK.monoidKRightIdentity(AP::just, EQ) },
-      Law("MonoidK Laws: Fold with Monoid instance") { SGK.monoidKFold(AP::just, EQ) })
-
-  fun <F> laws(SGK: MonoidK<F>, f: (Int) -> Kind<F, Int>, EQ: Eq<Kind<F, Int>>): List<Law> =
-    SemigroupKLaws.laws(SGK, f, EQ) + listOf(
-      Law("MonoidK Laws: Left identity") { SGK.monoidKLeftIdentity(f, EQ) },
-      Law("MonoidK Laws: Right identity") { SGK.monoidKRightIdentity(f, EQ) },
-      Law("MonoidK Laws: Fold with Monoid instance") { SGK.monoidKFold(f, EQ) })
-
-  fun <F> MonoidK<F>.monoidKLeftIdentity(f: (Int) -> Kind<F, Int>, EQ: Eq<Kind<F, Int>>): Unit =
-    forAll(Gen.int().map(f)) { fa: Kind<F, Int> ->
+    laws(SGK, Gen.int().map { AP.just(it) }, EQ)
+
+  fun <F> laws(SGK: MonoidK<F>, GEN: Gen<Kind<F, Int>>, EQ: Eq<Kind<F, Int>>): List<Law> =
```

Comment:
Same as above, either remove it or make it internal

Reply:
done